Modules/Loadable/Transforms/qSlicerTransformsModuleWidget.cxx

```diff
@@ -429,39 +420,12 @@ void qSlicerTransformsModuleWidget::pasteTransform()
 
   vtkNew<vtkMatrix4x4> tempMatrix;
 
-  QString text = QApplication::clipboard()->text();
-  QRegExp rx("(\\ |\\,|\\:|\\t|\\n|\\[|\\])");
-  QStringList entries = text.split(rx, QString::SkipEmptyParts);
-
-  if (entries.count() != 4
-      && entries.count() != 9
-      && entries.count() != 16)
+  std::string text = QApplication::clipboard()->text().toStdString();
+  bool success = vtkAddonMathUtilities::FromString(tempMatrix.GetPointer(), text);
+  if (success)
```

Comment:
In case of failure log a warning and in the message include the string that was not converted.